### ISSUE TITLE
🐛 `ProlateSpheroidal3D` metric is all-NaN on the entire ν = 0 plane

### ISSUE THREAD

**Issue by nstarman** [GalacticDynamics/coordinax#837]
Every point with `z = 0` — the galactic plane, the most common case for this chart — gives an all-NaN metric, and `check_data` explicitly accepts those points.

## Reproducer

```python
import jax; jax.config.update("jax_enable_x64", True)
import numpy as np, unxt as u, coordinax.charts as cxc, coordinax.manifolds as cxm

ch = cxc.ProlateSpheroidal3D(Delta=u.Q(2., "kpc"))
p = cxc.pt_map({"x": u.Q(3., "kpc"), "y": u.Q(4., "kpc"), "z": u.Q(0., "kpc")}, cxc.cart3d, ch)
# {"mu": 29.0, "nu": 0.0, "phi": 0.927}
ch.check_data(dict(p))                                  # accepts
cxc.pt_map(p, ch, cxc.cart3d)                           # round-trips exactly
cxm.metric_matrix(cxm.R3, dict(p), ch).diagonal         # [nan, nan, nan]
```

`scale_factors` and `norm` are NaN there too.

## Only one of the three entries is genuinely singular

Approaching ν → 0:

| ν | g_μμ | g_νν | g_φφ |
|---|---|---|---|
| 1e-4 | 9.99997e-03 | 1.8125e+04 | 24.99938 |
| 1e-8 | 1.00000e-02 | 1.8125e+08 | 24.99999 |
| **0** | **nan** | **nan** | **nan** |

`g_νν` is legitimately singular — ν is a degenerate coordinate on the equatorial disc. But `g_μμ = 0.01 kpc⁻²` and `g_φφ = ρ² = 25 kpc²/rad²` are finite and well defined, and are lost to a `0/0`.

## Cause

`src/coordinax/_src/euclidean/register_metric.py:477-497` takes `J = jac_pt_map(...)` then `DiagonalMetric((J.T @ J).diag())`. The `0/0` originates in `z = sqrt(mu*nu_D2)*sign(nu)` at `src/coordinax/_src/charts/register_ptmap.py:778` and `:1188`, giving `∂z/∂μ = (|ν|/Δ²)/(2√(μ|ν|/Δ²))`.

`|ν| = Δ²` (the z-axis) gives `[nan, inf, nan]` for the same reason.

The `check_data` domain (`mu ≥ Δ²`, `|nu| ≤ Δ²`) admits ν = 0, so nothing warns.

Found during an audit; independently reproduced.
